src/platform/plugins/shared/share/public/components/export_popover/export_popover.tsx

```diff
@@ -0,0 +1,316 @@
+  const [isCreatingExport, setIsCreatingExport] = useState<boolean>(false);
+  const [selectedMenuItemId, setSelectedMenuItemId] = useState<string>();
+  const selectedMenuItem = useMemo<ExportShareConfig | null>(() => {
+    return shareMenuItems.find((item) => item.id === selectedMenuItemId) ?? null;
+  }, [shareMenuItems, selectedMenuItemId]);
+  const [usePrintLayout, setPrintLayout] = useState(false);
+
+  const getReport = useCallback(async () => {
+    try {
+      setIsCreatingExport(true);
+      await selectedMenuItem?.config.generateAssetExport({
+        intl,
+        optimizedForPrinting: usePrintLayout,
+      });
+    } finally {
+      setIsCreatingExport(false);
+      onClose?.();
+    }
+  }, [intl, onClose, selectedMenuItem?.config, usePrintLayout]);
+
+  useEffect(() => {
+    // when there is only one share menu item,
+    // we want to open the flyout and not the popover
+    if (shareMenuItems.length === 1) {
+      setSelectedMenuItemId(shareMenuItems[0].id);
+      setIsFlyoutVisible(true);
+    }
+  }, [shareMenuItems]);
+
+  const flyoutOnCloseHandler = useCallback(() => {
+    return shareMenuItems.length === 1 ? onClose() : setIsFlyoutVisible(false);
+  }, [onClose, shareMenuItems.length]);
+
+  return (
+    <Fragment>
+      <EuiWrappingPopover
+        isOpen={!isFlyoutVisible}
+        data-test-subj="exportPopover"
+        button={anchorElement!}
+        closePopover={onClose}
+        panelPaddingSize="none"
+      >
+        <EuiListGroup>
+          {shareMenuItems.map((menuItem) => (
+            <EuiToolTip
+              position="left"
+              content={selectedMenuItem?.config.toolTipContent}
+              key={menuItem.id}
+            >
+              <EuiListGroupItem
+                iconType={menuItem.config.icon}
+                key={menuItem.id}
+                label={menuItem.config.label}
+                data-test-subj={`exportMenuItem-${menuItem.config.label}`}
+                isDisabled={menuItem.config.disabled}
+                onClick={() => {
+                  setSelectedMenuItemId(menuItem.id);
+                  setIsFlyoutVisible(true);
+                }}
+              />
+            </EuiToolTip>
+          ))}
+        </EuiListGroup>
+      </EuiWrappingPopover>
+      {isFlyoutVisible && (
+        <EuiFlyout
+          data-test-subj="exportItemDetailsFlyout"
+          size="s"
+          onClose={flyoutOnCloseHandler}
+          css={() => ({
+            ['--euiFixedHeadersOffset']: 0,
+          })}
+          ownFocus
+          maskProps={{
+            headerZindexLocation: 'above',
+          }}
+        >
+          <EuiFlyoutHeader hasBorder>
+            <EuiTitle>
+              <h2>
+                <FormattedMessage
+                  id="share.export.flyoutTitle"
+                  defaultMessage="Export {objectType} as {type}"
+                  values={{
+                    objectType: (objectTypeAlias ?? objectType).toLocaleLowerCase(),
```

Comment:
```suggestion
                    objectType: objectTypeAlias ?? objectType.toLocaleLowerCase(),
```

to don't apply low case for the custom `objectTypeAlias`

Reply:
I added this, so we don't have a situation that reads like so `Export Discover Session as PDF` in the case where objectTypeAlias provided is `Discover Session` is , so we consistently have `Export discover session as PDF` or `Export dashboard as PDF` following the [writing guidelines](https://eui.elastic.co/docs/content/writing/guidelines/index.html#capitalization) EUI provides.